model/src/main/java/com/redhat/ipaas/model/integration/Integration.java

```diff
@@ -65,6 +65,8 @@ default Kind getKind() {
 
     Optional<Status> getCurrentStatus();
 
+    Optional<List<String>> getStepsDone();
```

Comment:
I think these should be two different objects, and the processing parts (like 'desiredStatus' and 'stepsDone') should go into a separate object.